### DISCUSSION TITLE
Airflow taskinstance logs super verbose

### DISCUSSION THREAD

**Discussion by JeremieDoctrine** [apache/airflow#31926] [General]
### Apache Airflow version

Other Airflow 2 version (please specify below)

### What happened

Version 2.5.3

We use `KubernetesPodOperator` and Airflow logs are super verbose when an error occurs.

It dumps all the pod definitions in a multiline format and is not super useful for developers.

I was wondering how I could automatically filter out `taskinstance.py` logs from the interface

<img width="677" alt="CleanShot 2023-06-15 at 10 06 01@2x" src="https://github.com/apache/airflow/assets/59561370/fcac2a33-d44a-439e-9fd2-d341e611bdc3">

### What you think should happen instead

I should be able to automatically exclude the logs from the taskinstance.py `KubernetesPodOperator`

### How to reproduce

Trigger an error on the 

### Operating System

helm

### Deployment

Official Apache Airflow Helm Chart

### Are you willing to submit PR?

- [ ] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/apache/airflow/blob/main/CODE_OF_CONDUCT.md)

**Comment by potiuk** [comment]
This is not an issue, but discussion (when you open an issue it even tells you to open discussions in such cases).

**Comment by potiuk** [comment]
Follow the lead from https://airflow.apache.org/docs/apache-airflow/stable/administration-and-deployment/logging-monitoring/logging-tasks.html#advanced-configuration  and see if that is something that you can use for your case - also look at the various remote logging integration we have. Airlfow is already providing a lot of facilities so that you can change the way how looging is done, and what you get by default is considered as a starting point that you can freely configure as you see fit. 

There are various users with various expectations for the logging that will be usable for them (some of the claims you have above I would personally disagree with - but that's not a fact that's personal preference). But luckily we do not have to decide or change - you are in full control of configuring your logging as you want, you can even provide your own log handler filtering formatting etc. behaviour, following standard logging patterns of Python logging framework.

So if you want to convert multi-line exceptions into single line - feel free to use your formatter, if you do not want to log the exception, you can also skip it, or when you use remote logging, the multiline logging might be perfectly fine.

**Comment by JeremieDoctrine** [comment]
Hi, thanks potiuk for the answer.

I've read this documentation but it's not super clear how should I proceed. I've stumbled upon this blogs that does what I want : https://ashish.ch/adding-a-custom-filter-to-airflow-logs/ but I'm quite sure this is not an acceptable solution (patching the full file).

Do you have any general guidance on how to do this ?

**Comment by JeremieDoctrine** [comment]
OK i've managed to make it work ! Thanks for the guidance
